### PR TITLE
chore: Add `NewFilesOwnerID` parameter to deprovision sample script

### PR DESCRIPTION
**Pull request by congminh1254**
Closes: [SDK-2501](https://jira.inside-box.net/browse/SDK-2501)

- Only create folder when need transfer content.
- Use param `NewFilesOwnerID` to specify the folder ID which content will be moved to.

### CODE REVIEW

File: examples/User Deprovisioning/Users_Deprovision.ps1 [hunk 1]
```diff
@@ -6,9 +6,10 @@
 ########################################################################################
 
 param (
-    [switch]$DryRun = $false # if enabled, then no delete/create/update calls will be made, only read ones
+    [switch]$DryRun = $false, # if enabled, then no delete/create/update calls will be made, only read ones
+    [string]$NewFilesOwnerID = "" # The ID of the user to transfer files to before deleting the user
 )
-
+Set-Alias box /Users/mcong/boxcli/bin/run;
```
Comment (mgrytsai) [inline examples/User Deprovisioning/Users_Deprovision.ps1:12]: Looks like this part isn't needed for the script and this causes an error when you run script:

```
PS /Users/mgrytsai/IdeaProjects/boxcli/examples/User Deprovisioning> ./Users_Deprovision.ps1
box: /Users/mgrytsai/IdeaProjects/boxcli/examples/User Deprovisioning/Users_Deprovision.ps1:118
Line |
 118 |          return "$(box  configure:settings --help)" -like "*--analytic …
     |                    ~~~
     | The term '/Users/mcong/boxcli/bin/run' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included,
     | verify that the path is correct and try again.

box: /Users/mgrytsai/IdeaProjects/boxcli/examples/User Deprovisioning/Users_Deprovision.ps1:118
Line |
 118 |          return "$(box  configure:settings --help)" -like "*--analytic …
     |                    ~~~
     | The term '/Users/mcong/boxcli/bin/run' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included,
     | verify that the path is correct and try again.

PS /Users/mgrytsai/IdeaProjects/boxcli/examples/User Deprovisioning> ./Users_Deprovision.ps1
Importing csv of users to deprovision.
Successfully created new 'Employee Archive' root folder
```

File: examples/User Deprovisioning/Users_Deprovision.ps1 [hunk 2]
```diff
@@ -179,51 +193,80 @@ Function Start-Deprovisioning-Script {
...
-                Write-Log "Could not create new '$EmployeeArchiveFolderName' root folder. See log for details." -errorMessage $EmployeeArchiveFolderResp -output true -color Red
+                Write-Log "Could not get the current user. See log for details." -errorMessage $UserResp -output true -color Red
+                break
+            }
+        }
+        
+        # Check if user ID is valid
+        if ($UserId) {
+            try {
+                $UserResp = "$(box users:get --as-user=$UserId --json 2>&1)"
```
Comment (mgrytsai) [inline examples/User Deprovisioning/Users_Deprovision.ps1:225]: When running it as current user (not providing `NewFilesOwnerID`), I get error: 
> Could not get the user with ID 18057416283. See log for details.
> Unexpected API Response [403 Forbidden | dbqcbfh68lep41n2.0d6fa578496c7f1e78aacf6a4a76b92ab] access_denied_insufficient_permissions - Access denied - insufficient permission

I tried calling the raw command:
`box users:get --as-user=18057416283` (18057416283 is my current user id) and get above error, but running `box users:get ` without any issue

File: examples/User Deprovisioning/Users_Deprovision.ps1 (hunk 1 above)
Comment (congminh1254) [reply examples/User Deprovisioning/Users_Deprovision.ps1:12]: Ahhh, this one for debug only :'(

File: examples/User Deprovisioning/Users_Deprovision.ps1 (hunk 2 above)
Comment (congminh1254) [reply examples/User Deprovisioning/Users_Deprovision.ps1:225]: Maybe your scope is missing the option "Make API calls using the as-user header" so it's not allowing you to run command as another user.

Comment (mgrytsai) [reply examples/User Deprovisioning/Users_Deprovision.ps1:225]: No, it works when I provide another user ID and everything passes. But when i don't specify `NewFilesOwnerID`, so it uses current user id.

**Review by antusus — APPROVED** [review]
It is OK but I think we can reduce one API call to get user

File: examples/User Deprovisioning/Users_Deprovision.ps1 [hunk 3]
```diff
@@ -179,51 +193,89 @@ Function Start-Deprovisioning-Script {
...
+            }
+        }
+        
+        # Check if user ID is valid
+        if ($UserId -and ($UserId -ne $CurrentUserId)) {
+            try {
+                $UserResp = "$(box users:get --json 2>&1)"
+                $User = $UserResp | ConvertFrom-Json
+                $CurrentUserId = $User.id
+                if (!($CurrentUserId -eq $UserId)) {
```
Comment (antusus) [inline examples/User Deprovisioning/Users_Deprovision.ps1:231]: Why not `if($CurrentUserId -ne $UserId)` ?

**Review by antusus — APPROVED** [review]
👍

**Pull request merged**